mode/soy/soy.js

```diff
@@ -125,6 +125,44 @@
       this.scope = scope;
     }
 
+    function expression(stream, state) {
+      let match;
+      if (stream.match(/[[]/)) {
+        state.soyState.push("list-literal");
+        state.lookupVariables = false;
+        return null;
+      } else if (stream.match(/map/)) {
+        state.soyState.push("map-literal");
+        return "keyword";
+      } else if (stream.match(/record/)) {
+        state.soyState.push("record-literal");
+        return "keyword";
+      } else if (stream.match(/([\w]+)(?=\()/)) {
+        return "variable callee";
+      } else if (match = stream.match(/^["']/)) {
+        state.soyState.push("string");
+        state.quoteKind = match;
```

Comment:
This issue already existed before this patch, but it seems that the code that actually uses `state.quoteKind` compares it to a string, which only works because of implicit array-to-string conversion. Might be cleaner to change this to `= match[0]`

Reply:
Yep, fixed! Javascript is a bit too forgiving sometimes...